desidlas/training/training.py

```diff
@@ -0,0 +1,397 @@
+    best_accuracy = 0.0
+    best_offset_rmse = 999999999
+    best_density_rmse = 999999999
+    test_accuracy = None
+    loss_value = None
+    record_file=open('1016snr5smnewtrain.txt','w')
+
+    with tf.Graph().as_default():
+        train_step_ABC, tb_summaries = build_model(hyperparameters,INPUT_SIZE,matrix_size)
+
+        with tf.device('/gpu:1'), tf.compat.v1.Session(config=config) as sess:
+            # Restore or initialize model
+            if load_filename is not None:
+                tf.train.Saver().restore(sess, load_filename+".ckpt")
+                
+            else:
+               
+                sess.run(tf.compat.v1.global_variables_initializer())
+
+            
+            summary_writer = tf.summary.create_file_writer(tblogs)
+
+            for i in range(training_iters):
+                # Grab a batch
+                batch_fluxes, batch_labels_classifier, batch_labels_offset, batch_col_density = train_dataset.next_batch(batch_size)
+                # Train
+                sess.run(train_step_ABC, feed_dict={t('x'):                batch_fluxes,
+                                                    t('label_classifier'): batch_labels_classifier,
+                                                    t('label_offset'):     batch_labels_offset,
+                                                    t('label_coldensity'): batch_col_density,
+                                                    t('keep_prob'):        dropout_keep_prob})
+
+                if i % 200 == 0:   # i%2 = 1 on 200ths iteration, that's important so we have the full batch pos/neg
+                    train_accuracy, loss_value, result_rmse_offset, result_loss_offset_regression, \
+                    result_rmse_coldensity, result_loss_coldensity_regression \
+                        = train_ann_test_batch(sess, np.random.permutation(train_dataset.fluxes.shape[0])[0:10000],
+                                               train_dataset.data, summary_writer=summary_writer)
+                       
+                    
+                    print("step {:06d}, classify-offset-density acc/loss - RMSE/loss      {:0.3f}/{:0.3f} - {:0.3f}/{:0.3f} - {:0.3f}/{:0.3f}".format(
+                          i, train_accuracy, float(np.mean(loss_value)), result_rmse_offset,
+                             result_loss_offset_regression, result_rmse_coldensity,
+                             result_loss_coldensity_regression))
+                    record_file.write("step {:06d}, classify-offset-density acc/loss - RMSE/loss      {:0.3f}/{:0.3f} - {:0.3f}/{:0.3f} - {:0.3f}/{:0.3f}\n".format(
+                          i, train_accuracy, float(np.mean(loss_value)), result_rmse_offset,
+                             result_loss_offset_regression, result_rmse_coldensity,
+                             result_loss_coldensity_regression))
+                    
+                # Run on test
+                if i % 5000 == 0 or i == training_iters - 1:
```

Comment:
Should 5000 be a parameter?

Reply:
I think this may not be necessary to set this as a parameter. This code will print the testing accuracy every 5000 steps, I think this value is fine for the training?